netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -227,6 +232,8 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
       logger.fine("Network channel is closed");
       goAwayStatus(goAwayStatus().augmentDescription("Network channel closed"));
       cancelPing();
+      // Close the encoder now to cancel any buffered streams.
+      encoder().close();
```

Comment:
This will cause two calls to close the encoder, one here and one in 

https://github.com/netty/netty/blob/master/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java#L183

an alternative fix is to alter netty to close the encoders before it notifies stream closure to each stream in that method